xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/internal/html/DefaultHTMLCleanerTest.java

```diff
@@ -327,6 +327,40 @@ void restrictedAttributesAndTags() throws Exception
                 + "</mi></math>");
     }
 
+    /**
+     * Verify comment handling in restricted mode.
+     */
+    @Test
+    void restrictedComments()
+    {
+        Map<String, String> parameters = new HashMap<>(this.cleanerConfiguration.getParameters());
+        parameters.put("restricted", "true");
+        this.cleanerConfiguration.setParameters(parameters);
+
+        assertHTML("<p><strong>Hello  World</strong></p>", "<strong>Hello <!-- a comment --> World</strong>");
+        assertHTML("", "<!--My favorite operators are > and <!-->");
+
+        // Actually, just the comment should be removed but due to erroneous parsing in HTMLCleaner, the whole thing
+        // is treated as a comment.
+        assertHTML("", "<!--> <a href=\"#\">no comment</a>");
+        assertHTML("", "<!---> <a href=\"#\">no comment</a>");
+    }
+
+    @Test
+    void comments()
+    {
+        assertHTML("<!--My favorite operators are > and <!-->", "<!--My favorite operators are > and <!-->");
```

Comment:
On small improvement I can think of is to use parameterized tests. The main interest it that the test will not stop when the first assertion fail, but will report individual failures for each assertions.

Reply:
I've changed the two new tests into parameterized tests.